opentelemetry-sdk/src/opentelemetry/sdk/_logs/_internal/__init__.py

```diff
@@ -673,6 +667,46 @@ def get_logger(
             ),
         )
 
+    @lru_cache(maxsize=None)
```

Comment:
Is `lru_cache` thread safe?

Reply:
It is thread safe (https://docs.python.org/3/library/functools.html#functools.lru_cache), but I actually realized the linter complains about `lru_cache` being used on methods with `self`, I switched it back to my initial approach of using a dict and threading.Lock. Using custom cache also caches at the instance level, whereas lru_cache is at the class level https://docs.python.org/3/faq/programming.html#faq-cache-method-calls

Sorry for the back and forth, but re-requesting review for this point here.

Reply:
No problem, thanks for your responses on the PR. It LGTM with the explicit approach